### PR TITLE
Clarify how request's mode and credentials mode are set (#609)

### PR DESCRIPTION
**Pull request by hayatoito**
Add a section which explains how request's mode and credentials mode
are set with `<link>`-based API.

### CODE REVIEW

File: explainers/subresource-loading.md [hunk 1]
```diff
@@ -154,6 +155,35 @@ the bundle, and a subframe is instantiated as an
 
 Note that `resources` attribute is reflected to JavaScript as a [`DOMTokenList`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList).
 
+## Request's mode and credentials mode
+
+With `<link>`-based API, a
+[request](https://fetch.spec.whatwg.org/#concept-request) for a bundle
+will have their [mode][request mode] set to "`no-cors`" and their
+[credentials mode][credentials mode] set to "`include`" unless a
+[crossorigin][crossorigin attribute] attribute is specified.
```
Comment (jyasskin) [inline explainers/subresource-loading.md:164]:
```suggestion
With the `<link>`-based API, a
[request](https://fetch.spec.whatwg.org/#concept-request) for a bundle
will have its [mode][request mode] set to "`no-cors`" and its
[credentials mode][credentials mode] set to "`include`" unless a
[crossorigin][crossorigin attribute] attribute is specified.
```

File: explainers/subresource-loading.md [hunk 2]
```diff
@@ -154,6 +155,35 @@ the bundle, and a subframe is instantiated as an
...
+
+If a [crossorigin][crossorigin attribute] attribute is specified,
+their [mode][request mode] and [credentials mode][credentials mode]
```
Comment (jyasskin) [inline explainers/subresource-loading.md:167]:
```suggestion
the request's [mode][request mode] and [credentials mode][credentials mode]
```

File: explainers/subresource-loading.md [hunk 3]
```diff
@@ -154,6 +155,35 @@ the bundle, and a subframe is instantiated as an
...
+will be set as the [CORS settings attributes][cors settings attribute]
+section describes.
+
+The following table is the summary.
+
+| State          | [mode][request mode] | [credentials mode][credentials mode] |
+| --------------- | -----------          | ----------------                     |
+| No CORS         | "`no-cors`"          | "`include`" [1]                       |
```
Comment (jyasskin) [inline explainers/subresource-loading.md:175]: Can we follow the ["For more modern features"](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#cors-settings-attributes:~:text=For%20more%20modern%20features) policy, and always use CORS for bundle fetching?

**Review by jyasskin — APPROVED** [review]

File: explainers/subresource-loading.md (hunk 1 above)
Comment (hayatoito) [reply explainers/subresource-loading.md]: Done. Thanks!

File: explainers/subresource-loading.md (hunk 2 above)
Comment (hayatoito) [reply explainers/subresource-loading.md]: Done. Thanks!

File: explainers/subresource-loading.md (hunk 3 above)
Comment (hayatoito) [reply explainers/subresource-loading.md]: As per the separate discussion, we've decided to follow "For more modern features" policy.

Now we use "cors" always, and use "same-origin" for a credentials mode if `crossorigin`attribute is not given.

Comment (jyasskin) [reply explainers/subresource-loading.md]: Thanks! And sorry for forgetting why we had `no-cors` in the first place.

**Review by jyasskin — APPROVED** [review]

**Comment by hayatoito** [comment]
Thanks for the review!

**Pull request merged**